SECURITY.md

```diff
@@ -0,0 +1,59 @@
+# Security Policy
+
+## Reporting a Vulnerability
+
+To report a vulnerability, either:
+
+1. Report it on Github directly you can follow the procedure described
+[here](https://docs.github.com/en/code-security/security-advisories/guidance-on-reporting-and-writing/privately-reporting-a-security-vulnerability)
+and:
+    - Navigate to the security tab on the repository
+    - Click on 'Advisories'
+    - Click on 'Report a vulnerability'
+
+2. Send an email to `security@crossplane.io` detailing the issue and steps
+to reproduce.
+
+The reporter(s) can expect a response within 24 hours acknowledging
+the issue was received. If a response is not received within 24 hours, please
+reach out to any committer directly to confirm receipt of the issue.
+
+## Review Process
+
+Once a committer has confirmed the relevance of the report, a draft security
+advisory will be created on Github. The draft advisory will be used to discuss
+the issue with committers, the reporter(s), and crossplane's security advisors.
+If the reporter(s) wishes to participate in this discussion, then provide
+reporter Github username(s) to be invited to the discussion. If the reporter(s)
+does not wish to participate directly in the discussion, then the reporter(s)
+can request to be updated regularly via email.
+
+If the vulnerability is accepted, a timeline for developing a patch, public
+disclosure, and patch release will be determined. If there is an embargo period
+on public disclosure before the patch release, an announcment will be sent to
+the security announce mailing list (`crossplane-security-announce@lists.cncf.io`)
```

Comment:
we'll need to follow up with the CNCF to get this email address created too.

Perhaps we can make the reporting email symmetrical with the announcement email?  e.g. let's just create both on lists.cncf.io?

Reply:
I'm not sure what's the procedure to handle such lists under the cncf, but while the announcement list is a public thing and people should be able to subscribe or be subscribed easily, `security@crossplane.io` should only send emails to a slowly changing set of people, maintainers only most probably, so, it might be easier to handle it having it totally under our control.

Reply:
No strong opinions on that though, Rook is also already using "symmetrical" [emails](https://github.com/rook/rook/blob/master/SECURITY.md#mailing-lists).

Reply:
I thought about this a bit more, and I think having all security lists hosted by the CNCF is the right thing to do.  Those lists can be completely restricted as we see fit, and we either have self-service to make those changes ourselves, or updates to the lists are a quick CNCF serviceDesk ticket away.

Reply:
@phisco I have opened a CNCF service desk ticket to get these mailing lists created!

https://cncfservicedesk.atlassian.net/servicedesk/customer/portal/1/CNCFSD-1597 (this link is probably not accessible for most folks)